pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -511,6 +511,7 @@ let
       # Support configuring jack functions via fw mechanism at boot
       SND_HDA_PATCH_LOADER = yes;
       SND_HDA_CODEC_CA0132_DSP = whenOlder "5.7" yes; # Enable DSP firmware loading on Creative Soundblaster Z/Zx/ZxR/Recon
+      SND_HDA_CODEC_CS8409 = module; # Cirrus Logic HDA Bridge CS8409
```

Comment:
I am still waiting for the build to finish but so far it looks like at least some of the  hardened kernel likes this option:

```
┃    ┌─ ⚠ linux-config-5.4.277 failed with exit code 1 after ⏱ 25s in buildPhase
┃ ┌─ ⏸ linux-hardened-5.4.277
┃ │  ┌─ ⚠ linux-config-5.10.218 failed with exit code 1 after ⏱ 23s in buildPhase
┃ ├─ ⏸ linux-hardened-5.10.218
┃ │  ┌─ ⚠ linux-config-4.19.315 failed with exit code 1 after ⏱ 19s in buildPhase
┃ ├─ ⏸ linux-hardened-4.19.315
┃ │  ┌─ ✔ linux-config-6.9.7 ⏱ 41s
┃ ├─ ⏵ linux-libre-6.9.7 (buildPhase) ⏱ 1h1m18s
┃ │  ┌─ ✔ linux-config-6.9.7 ⏱ 41s
┃ ├─ ⏵ linux-lqx-6.9.7 (buildPhase) ⏱ 1h1m20s
┃ ├─ ⏵ linux-zen-6.9.7 (buildPhase) ⏱ 1h1m43s
┃ ├─ ⏵ linux-hardened-6.8.11 (buildPhase) ⏱ 1h2m1s
┃ ├─ ⏵ linux-xanmod-6.6.33 (buildPhase) ⏱ 1h2m1s
┃ ├─ ⏵ linux-xanmod-6.9.4 (buildPhase) ⏱ 1h2m1s
┃ ├─ ⏵ linux-hardened-6.9.2 (buildPhase) ⏱ 1h2m1s
┃ ├─ ⏵ linux-6.9.7 (buildPhase) ⏱ 1h2m1s
┃ ├─ ⏵ linux-libre-6.6.36 (buildPhase) ⏱ 1h2m1s
┃ ├─ ⏵ linux-6.6.36 (buildPhase) ⏱ 1h2m1s
┃ ├─ ⏵ linux-hardened-6.1.92 (buildPhase) ⏱ 1h2m2s
┃ ├─ ⏵ linux-hardened-6.6.32 (buildPhase) ⏱ 1h2m2s
```

```
CHOICE: 1-3?, ANSWER:
GOT:
GOT: *
GOT: * Kernel Testing and Coverage
GOT: *
GOT: *
GOT: * KUnit - Enable support for unit tests
GOT: *
QUESTION: KUnit - Enable support for unit tests, NAME: KUNIT, ALTS: N/m/y/?, ANSWER: n
GOT: n
QUESTION: Notifier error injection, NAME: NOTIFIER_ERROR_INJECTION, ALTS: N/m/y/?, ANSWER: n
GOT: n
QUESTION: Fault-injections of functions, NAME: FUNCTION_ERROR_INJECTION, ALTS: N/y/?, ANSWER:
GOT:
QUESTION: Fault-injection framework, NAME: FAULT_INJECTION, ALTS: N/y/?, ANSWER:
GOT:
QUESTION: Code coverage for fuzzing, NAME: KCOV, ALTS: N/y/?, ANSWER:
GOT:
GOT: *
GOT: * Runtime Testing
GOT: *
QUESTION: Runtime Testing, NAME: RUNTIME_TESTING_MENU, ALTS: N/y/?, ANSWER: n
GOT: n
QUESTION: Memtest, NAME: MEMTEST, ALTS: Y/n/?, ANSWER: y
GOT: y
QUESTION: Microsoft Hyper-V driver testing, NAME: HYPERV_TESTING, ALTS: N/y/?, ANSWER:
GOT:
GOT: #
GOT: # configuration written to .config
GOT: #
GOT: make: Leaving directory '/build/linux-5.10.218'
warning: unused option: CFI_CLANG
warning: unused option: CONFIG_LDISC_AUTOLOAD
warning: unused option: DEBUG_INFO_BTF
warning: unused option: INET_DIAG_DESTROY
warning: unused option: INET_MPTCP_DIAG
warning: unused option: INET_RAW_DIAG
warning: unused option: INET_TCP_DIAG
warning: unused option: INET_UDP_DIAG
warning: unused option: IOMMU_DEFAULT_DMA_LAZY
warning: unused option: IOMMU_DEFAULT_DMA_STRICT
warning: unused option: IO_STRICT_DEVMEM
warning: unused option: POWER_RESET_GPIO
warning: unused option: POWER_RESET_GPIO_RESTART
warning: unused option: SECURITY_WRITABLE_HOOKS
error: unused option: SND_HDA_CODEC_CS8409
warning: unused option: UBSAN_LOCAL_BOUNDS
```

You should be able to reproduce this by building:

```
nix-build -A linuxKernel.packages.linux_5_10_hardened.kernel
```

Reply:
So these are all new regressions:

```
Result of `nixpkgs-review pr 322968` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_4_19_hardened.kernel</li>
    <li>linuxPackages_4_19_hardened.kernel.dev</li>
    <li>linuxPackages_5_10_hardened.kernel</li>
    <li>linuxPackages_5_10_hardened.kernel.dev</li>
    <li>linuxPackages_5_4_hardened.kernel</li>
    <li>linuxPackages_5_4_hardened.kernel.dev</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>linuxPackages-libre.kernel</li>
    <li>linuxPackages-libre.kernel.dev</li>
    <li>linuxPackages.kernel</li>
    <li>linuxPackages.kernel.dev</li>
    <li>linuxPackages_5_15_hardened.kernel</li>
    <li>linuxPackages_5_15_hardened.kernel.dev</li>
    <li>linuxPackages_6_1_hardened.kernel</li>
    <li>linuxPackages_6_1_hardened.kernel.dev</li>
    <li>linuxPackages_hardened.kernel (linuxPackages_6_6_hardened.kernel)</li>
    <li>linuxPackages_hardened.kernel.dev (linuxPackages_6_6_hardened.kernel.dev)</li>
    <li>linuxPackages_6_8_hardened.kernel</li>
    <li>linuxPackages_6_8_hardened.kernel.dev</li>
    <li>linuxPackages_6_9_hardened.kernel</li>
    <li>linuxPackages_6_9_hardened.kernel.dev</li>
    <li>linuxPackages_latest-libre.kernel</li>
    <li>linuxPackages_latest-libre.kernel.dev</li>
    <li>linuxPackages_latest.kernel</li>
    <li>linuxPackages_latest.kernel.dev</li>
    <li>linuxPackages_lqx.kernel</li>
    <li>linuxPackages_lqx.kernel.dev</li>
    <li>linuxPackages_xanmod.kernel</li>
    <li>linuxPackages_xanmod.kernel.dev</li>
    <li>linuxPackages_xanmod_latest.kernel (linuxPackages_xanmod_stable.kernel)</li>
    <li>linuxPackages_xanmod_latest.kernel.dev (linuxPackages_xanmod_stable.kernel.dev)</li>
    <li>linuxPackages_zen.kernel</li>
    <li>linuxPackages_zen.kernel.dev</li>
  </ul>
</details>
```

Reply:
These needs to be addressed before we can merge this.

Reply:
Can we skip the backport part? The kernel option is less than a year old ... 

https://www.kernelconfig.io/config_snd_hda_codec_cs8409

Reply:
No backport required but we only should set this options for kernel that have support for it.

Reply:
Just set an appropriate `versionAtLeast` constraint.

Reply:
kernel >=6.6 should be an safe option for official 24.05 supporting this hardware